rascal-lsp/src/main/rascal/demo/lang/pico/LanguageServer.rsc

```diff
@@ -36,17 +36,22 @@ module demo::lang::pico::LanguageServer
 import util::LanguageServer;
 import util::IDEServices;
 import ParseTree;
+import util::ParseErrorRecovery;
 import util::Reflective;
 import lang::pico::\syntax::Main;
 
+private Tree picoParser(str input, loc origin) {
+    return parse(#start[Program], input, origin, allowRecovery=true, filters={createParseErrorFilter(false)});
+}
+
 @synopsis{A language server is simply a set of ((LanguageService))s.}
 @description{
 Each ((LanguageService)) for pico is implemented as a function.
 Here we group all services such that the LSP server can link them
 with the ((Language)) definition later.
 }
 set[LanguageService] picoLanguageServer() = {
-    parsing(parser(#start[Program]), usesSpecialCaseHighlighting = false),
+    parsing(picoParser, usesSpecialCaseHighlighting = false),
```

Comment:
Remind me: now we're depending on the evaluator lock right? There was no current way around it? (because we're calling the `filters`)

Reply:
That is true. But we have discussed in the past that it might be possible to recognize this specific situation: when calling the parser with only the error disambiguation filter we can forfeit calling the evaluator and do the filtering at the Java level.